go/response_handlers.go

```diff
@@ -1776,3 +1777,103 @@ func handleSortedSetWithScoresResponse(response *C.struct_CommandResponse, rever
 
 	return zRangeResponseArray, nil
 }
+
+func handleXInfoStreamResponse(response *C.struct_CommandResponse) (models.XInfoStreamResponse, error) {
+	defer C.free_command_response(response)
+
+	typeErr := checkResponseType(response, C.Map, false)
+	if typeErr != nil {
+		return models.XInfoStreamResponse{}, typeErr
+	}
+
+	result, err := parseMap(response)
+	if err != nil {
+		return models.XInfoStreamResponse{}, err
+	}
+
+	infoMap, ok := result.(map[string]any)
+	if !ok {
+		return models.XInfoStreamResponse{}, &errors.RequestError{
+			Msg: fmt.Sprintf("unexpected type for stream info: %T", result),
+		}
+	}
+
+	streamInfo := models.XInfoStreamResponse{}
+
+	// Parse integer fields
+	if val, ok := infoMap["length"].(int64); ok {
```

Comment:
should we return an error if a mandatory field is missing or has incorrect format?